### DISCUSSION TITLE
Galactic on create3 and humble on linux pc?

### DISCUSSION THREAD

**Discussion by framlin** [iRobotEducation/create3_docs#258] [Questions] [answered]
I am totally new on ros and ros 2 and have ordered a create 3 to learn it 2 days ago.
Is it possible, to use the create3 running galactic together with a linux pc running humble over wifi?
Or is it preferable, to run the humble beta release on create 3 instead.

**Comment by alsora** [comment, marked as answer]
The recommended approach is to run galactic on both sides.

You could use a docker image with galactic if this is not available as Debian package for your Linux distribution (we have create 3 dedicated docker images) see instructions here https://iroboteducation.github.io/create3_docs/setup/docker/

If galactic is not an option for some reasons, then yes, you can use the humble beta release on the robot.

Running mixed releases (galactic on robot and Humble on desktop) should be your last option. 
There may be subtle issues running mixed releases, so I wouldn't recommend that to a beginner (it is not officially supported but there are people doing it).
Moreover, something to keep in mind is that there is a difference in the create 3 APIs between galactic and Humble (we renamed some messages relate to the dock). 
Make sure to use the version of the irobot_create_msgs that is required by your robot (this is defined in the firmware release page and its 1.2.4 for the galactic robots; this is not the main branch and it's not what you would install with apt-get install with ros Humble, so you will have to build this version from sources) https://iroboteducation.github.io/create3_docs/releases/g_3_1/

**Comment by framlin** [comment]
Thank you very much for your answer!
